test/swoosh/adapters/amazonses_test.exs

```diff
@@ -154,6 +164,46 @@ defmodule Swoosh.Adapters.AmazonSESTest do
     assert AmazonSES.deliver(email, config) == {:error, %{code: "ErrorCode", message: "Error Message"}}
   end
 
+  @tag capture_log: true
+  test "a sent email that returns an empty api error", %{bypass: bypass, config: config, valid_email: email} do
+    Bypass.expect bypass, fn conn ->
+      conn = parse(conn)
+      expected_path = "/"
+
+      assert expected_path == conn.request_path
+      assert "POST" == conn.method
+
+      Plug.Conn.resp(conn, 500, @empty_error_response)
+    end
+
+    assert AmazonSES.deliver(email, config) ==
+      {:error, %{code: "ServiceUnavailable", message: "Empty response from SES"}}
+
+    assert capture_log(fn ->
+      Logger.error("Empty response with code: 500")

```

Comment:
This doesn't test the code. You're logging it out manually and then asserting the same thing. Instead, the capture_log should be around `AmazonSES.deliver` for it to be a real test.

Reply:
Ok, maybe now 😄

Reply:
That makes more sense now :+1: Lets deploy this and wait for the next occurrence. Then I think we can simplify it and make an upstream PR as well.